manifests/params.pp

```diff
@@ -10,7 +10,7 @@
   $package_ensure    = 'latest'
   $ui_package_name   = 'consul_ui'
   $ui_package_ensure = 'latest'
-  $version = '0.5.0'
+  $version = hiera('consul::version', '0.5.0')  # ensure defaults resolve from hiera as well
```

Comment:
```$consul::version``` works fine
but in init.pp we have defaults:
```
$download_url = "https://dl.bintray.com/mitchellh/consul/${version}_${os}_${arch}.zip",
$ui_download_url = "https://dl.bintray.com/mitchellh/consul/${version}_web_ui.zip",
```
These defaults apparently do *not* get populated by hiera. Eg if you have in hiera:
```consul::version: 0.5.1``` and in params.pp ```$version = '0.5.0'``` this means your url will become ```https://dl.bintray.com/mitchellh/consul/0.5.0_web_ui.zip``` instead of 0.5.1 (which is used in other places because init.pp:version is correctly resolved from hiera)